CODE/Sources/PennyBOT/Services/DiscordService/DiscordService.swift

```diff
@@ -376,6 +376,12 @@ actor DiscordService {
         })
     }
 
+    func memberHasRolesForElevatedRestrictedCommandsAccess(member: Guild.Member) -> Bool {
+        Constants.Roles.elevatedRestrictedCommandsAccess.contains(where: {
+            member.roles.contains($0.rawValue)
+        })
```

Comment:
```suggestion
        !(Set(member.roles).intersection(
            Constants.Roles.elevatedRestrictedCommandsAccess.map(\.rawValue)
        )).isEmpty
```

Reply:
Is it worth going through the whole hash-map creation just to perform the containment check? 🤔

Reply:
Yes indeedy. It's actually faster this way even for tiny arrays - believe me, I've checked.

Proof:

1. `Set.init(_:)` and `Collection.map(_:)` are both `O(n)`, as is `Set.intersection(_:)` where the input is a `Sequence`. Big-O factors out coefficients, so the entire thing is effectively `O(n)`.
2. `Collection.contains(where:)` is `O(n)`, so nesting it yields `O(n*m)` (where `m` represents the second array). Big-O also disregards such differences (being somwhat similar to a derivative), which gives you `O(n^2)` complexity.

Even for linear versus polynomial complexity, Are you gonna notice the difference when the inputs are arrays of sizes 4 and 6 (just for example)? Very unlikely, since string comparison usually reduces to `memcmp()` when the length check doesn't short-circuit in modern Swift. But, the nested-contains thing is an _EXTREMELY_ common anti-pattern I see people using all the time, and quite often in cases where it makes a _VERY_ large difference (for example, say the inner loop was executing database queries on each iteration - which I've run into several times in production code!). It is absolutely worth the effort even if just to get yet another example of what not to do out of opensource code.